### ISSUE TITLE
[Bug] WebAuthenticator on Android in Release mode only hangs on return to app

### ISSUE THREAD

**Issue by michaeldimoudis** [xamarin/Essentials#1745]
### Steps to Reproduce

1. Implement WebAuthenticator in a Xamarin Forms app and run it on a device to log in
2. The callback URL should be something like myapp://callback
3. As per docs the Android Intent Filter is:
```
[Activity(NoHistory = true, LaunchMode = LaunchMode.SingleTop)]
[IntentFilter(new[] { Intent.ActionView },
    Categories = new[] { Intent.CategoryDefault, Intent.CategoryBrowsable },
    DataScheme = "myapp",
    DataHost = "callback")]
public class WebAuthenticationCallbackActivity : Xamarin.Essentials.WebAuthenticatorCallbackActivity
{
}
```

### Expected Behavior
Will be that the PKCE flow will open the app so it can exchange the code for an access token.
This works in an Android simulator on Windows using Android 7.1 and 9.0. (Code also works on an iOS simulator and iOS device)

### Actual Behavior
Actual behaviour when either pressing the cross to cancel the auth flow, or a successful callback in to the app, the app hangs. The background color of the app only appears on the screen, so in my case the screen is just a solid light blue.

I've tried everything for the past month trying to resolve it. Tried API 29 and 30 (I did add `android.support.customtabs.action.CustomTabsService` in manifest), apk and bundle, various different builds settings.

When I build the app in debug mode and run it on a device, it works. However when I build the app in release mode, either in the simulator or device it hangs. Unfortunately I can't get any debug information out as it's in release mode and nothing comes up in Visual Studio on F5.

Would be great if someone knows anything about this issue I'm having. Thanks in advance...

### Basic Information

- Version with issue: A real Android device, tested on 6.0 and 7.1.2
- Last known good version: n/a
- IDE: VS
- Platform Target Frameworks: 
  - Android: API 21 to 29/30
- Affected Devices: Works on a simulator in debug, does not work on a real device deployed through Google Play beta testing.

**Comment by BenFidge** [comment]
I'm seeing the same on Android 10 device. Error sent to MS App Center is:

`System.NullReferenceException: Object reference not set to an instance of an object`


Stack Trace:
```
WebAuthenticatorIntermediateActivity.OnCreate (Android.OS.Bundle savedInstanceState)
Activity.n_OnCreate_Landroid_os_Bundle_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_savedInstanceState)
(wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.7(intptr,intptr,intptr)
```

Have also played with different target frameworks, and added `android.support.customtabs.action.CustomTabsService` to manifest, but no joy.

**Comment by michaeldimoudis** [comment]
Phew I'm glad I'm not the only one with the issue. Been trying different things on and off for a month. Was about to start pulling everything out, nuget packages and additional initialisations. 

Anyone know of an alternative solution? What was the go to before Xamarin Essentials so I can give that a try...? 🤔

**Comment by BenFidge** [comment]
Hi Michael - i went into Settings and set Chrome as default browser and its all working for me now, probably because this supports tabs, whereas "Browser" doesn't i dont think. I bought this device today so its more or less factory settings. 

I've set my target to Min Android version to 8.0 (API 26), Target to 10.0 (Api 29) and Compile using 10.0 too btw

Its a bit of a worry that we have a dependency on Chrome. Maybe the fix coming in Xamarin Essential 1.6.2 [PR 1687](https://github.com/xamarin/Essentials/pull/1687) might alleviate this

**Comment by michaeldimoudis** [comment]
Thanks for the heads up Ben. Hope that commit is released soon. It seems my issue might be different. I do have Chrome as the default browser. The auth and flow works great in debug mode in sim and device, however when it's built for release mode it does not work in sim nor device. Hence it's such a head scratcher. I'm going to give Xamarin.Auth.XamarinForms nuget a try and see if I can get that to work.

**Comment by BenFidge** [comment]
Not heard of that, let us know how you get on. One thing you can try is take this repo, build it locally and use that DLL see if it works for you

**Comment by michaeldimoudis** [comment]
OMG I spent a month on and off looking at this issue. I tried the sample here https://github.com/IdentityModel/IdentityModel.OidcClient.Samples/blob/main/XamarinForms/README.md with the same issue exhibiting in my app, so it must be us. After I stripped all my library initializations and added each back one by one, I tracked the issue down to a single line where we flushed analytics data on `OnSleep()`, it exhibited weird behavior in release mode only causing the app to crashing coming back in. Geez. I guess the good thing to come out of this was cleaning up a bunch of legacy code in hope it solved the issue. Closing this now as it's obviously not Xamarin.Essentials. Thanks for the chat which prompted me to investigate further.
